dataland-frontend/src/components/resources/frameworkDataSearch/FrameworkDataSearchBar.vue

```diff
@@ -126,37 +140,43 @@ export default defineComponent({
     searchBarName() {
       this.autocomplete.focus();
     },
+    filter: {
+      handler() {
+        this.searchBarInput = this.filter?.companyNameFilter ?? "";
+        this.queryCompany();
+      },
+      deep: true,
+    },
   },
 
   data: function () {
     return {
+      searchBarInput: "",
       autocompleteArray: [] as Array<object>,
       autocompleteArrayDisplayed: [] as Array<object>,
       loading: false,
       route: useRoute(),
     };
   },
-
   methods: {
-    handleInput(inputEvent: { target: { value: string } }) {
-      this.$emit("update:modelValue", inputEvent.target.value);
-    },
-
     handleItemSelect(event: { value: DataSearchStoredCompany }) {
-      this.$router.push(this.getRouterLinkTargetFrameworkInt(event.value));
+      this.$router.push(getRouterLinkTargetFramework(event.value));
     },
     handleKeyupEnter() {
-      this.queryCompany();
+      this.$emit("search-confirmed", this.searchBarInput);
       this.autocomplete.hideOverlay();
       this.autocomplete.$refs.input.blur();
+      this.queryCompany();
     },
     async queryCompany() {
-      if (this.getKeycloakPromise !== undefined) {
+      if (this.getKeycloakPromise !== undefined && this.enableFullSearch) {
```

Comment:
What is the point of the "enableFullSearch" boolean? It is only used once in our code and set to true.
In which cases should a search not be full?

Reply:
This variable has a default value of "false" and is not set on the view framework data pages. I introduced it to prevent the second search API call on these pages as the result is never used.

Reply:
Got it.
Renamed the variable to "emitSearchResultsArray" so one can understand, that no search results array is provided by the component if the flag is set to "false" (which is default).